common/src/main/java/io/quarkiverse/operatorsdk/common/Files.java

```diff
@@ -0,0 +1,165 @@
+
+    public static boolean generateAPIFiles(String group, String version, String kind, MessageWriter messager) {
+        final var packageName = reversePeriodSeparatedString(group);
+        final var finalKind = capitalizeIfNeeded(kind);
+
+        final var targetDir = ROOT_TARGET_DIR + packageName.replace('.', '/') + "/";
+        try {
+            java.nio.file.Files.createDirectories(Paths.get(targetDir));
+        } catch (IOException e) {
+            messager.error("Couldn't create target directory " + targetDir, e);
+            return false;
+        }
+
+        final var apiTemplatesDir = Thread.currentThread().getContextClassLoader().getResource(API_TEMPLATES_PATH);
+        if (apiTemplatesDir == null) {
+            messager.error("Couldn't find " + API_TEMPLATES_PATH + " directory in resources");
+            return false;
+        }
+
+        // make sure we can get the files from a jar, which, for some reason, doesn't work directly
+        // see https://stackoverflow.com/questions/22605666/java-access-files-in-jar-causes-java-nio-file-filesystemnotfoundexception
+        final URI apiTemplatesDirURI;
+        try {
+            apiTemplatesDirURI = apiTemplatesDir.toURI();
+        } catch (URISyntaxException e) {
+            messager.error("Couldn't convert " + apiTemplatesDir + " path to URI", e);
+            return false;
+        }
+        if ("jar".equals(apiTemplatesDirURI.getScheme())) {
+            for (FileSystemProvider provider : FileSystemProvider.installedProviders()) {
+                if (provider.getScheme().equalsIgnoreCase("jar")) {
+                    try {
+                        provider.getFileSystem(apiTemplatesDirURI);
+                    } catch (FileSystemNotFoundException e) {
+                        // if for some reason, there's no file system for our jar, initialize it first
+                        try {
+                            provider.newFileSystem(apiTemplatesDirURI, Collections.emptyMap());
+                        } catch (IOException ex) {
+                            messager.error("Couldn't initialize FileSystem instance to read resources", e);
+                            return false;
+                        }
+                    }
+                }
+            }
+        }
+
+        final var data = Map.<String, Object> of("packageName", packageName,
+                "group", group,
+                "version", version,
+                "kind", finalKind);
+
+        final var apiTemplatesPath = Paths.get(apiTemplatesDirURI);
+        try (var paths = java.nio.file.Files.walk(apiTemplatesPath)) {
+            paths.filter(f -> java.nio.file.Files.isRegularFile(f) && f.getFileName().toString().endsWith(".qute"))
+                    .forEach(path -> {
+                        try {
+                            final var templateAsString = java.nio.file.Files.readString(path);
+                            final var templateInstance = Qute.fmt(templateAsString)
```

Comment:
Like I mentioned, this is not ideal in the sense that the template content is parsed every time the `Qute#fmt()` method is called. If you need to render the template multiple times then either use `Qute.enableCache()` or Qute.engine().parse(...)` together with `Qute.engine().putTemplate()` and `Qute.engine().getTemplate()`...

Reply:
What I'm saying is: if you only render the template once, then `Qute.fmt(templateAsString)` is OK. If multiple times, then either use `Qute.enableCache()` - this enables cache for `Qute.fmt()` by default (`templateAsString` will be parsed once); or use something like:
```java
// This should be called once
Engine engine = Qute.engine();
engine.putTemplate("myTemplate", engine.parse(templateAsString));
...
// And then to render a template:
Qute.engine().getTemplate("myTemplate").render();
```

Reply:
Ah, I think I was missing the `putTemplate` part, I thought that I could call `getTemplate` directly and that wasn't working. Thanks! That said, since these things won't probably be run very often, I'm not sure it matters but I will still make the change (it might help when called from the dev console, maybe?). Then again, you could make a point that if it's not going to be call often, caching a template might actually be worse because then you commit memory to something that will only be used infrequently and possibly only once.

Reply:
Yes, it really depends on the use case :shrug: .